src/metabase/driver.clj

```diff
@@ -149,6 +149,12 @@
      *  `:set-timezone`
      *  `:standard-deviation-aggregations`")
 
+  (field-avg-length ^Integer [this, ^FieldInstance field]
+    "*OPTIONAL*. Return the avergage length of FIELD. Default implementation does this in Clojure-land.")
+
+  (field-percent-urls ^Double [this, ^FieldInstance field]
```

Comment:
Any functionality in drivers (including sync functionality) that can be overriden should be a part of the `IDriver` protocol. That way drivers can swap out a faster implementation if they have one or pop in a no-op implementation if they want to skip the step altogether. It's just as easier, if not easier, to swap out implementations this way (easier, IMO, because the method is documented in a place you'd know to look)

It makes things easier for everyone to have a single formal protocol instead of a single formal method and a handful of informal protocol. It's what's documented in our [Guide to Writing a Driver](https://github.com/metabase/metabase/wiki/Writing-A-Driver)